### PR TITLE
Add google/rpc code

### PR DESCRIPTION
**Pull request by douglas-reid**
Provides gogoslick version of `google.golang.org/genproto/googleapis/google/rpc/...` repository.
- [x] Makefile for updating generated code
- [x] Use dep to pin gogo binaries (+ SHA in Makefile to pin source)
- [x] Record version of `protoc` alongside files
- [x] Uses latest protoc release (3.5.0)

Also adds some minimal documentation on usage.

### CODE REVIEW

File: googleapis/LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (geeknoid) [inline googleapis/LICENSE]: I think this file should be in the root of the repo.

**Comment by kyessenov** [comment]
Yeah you want to capture generation with a script and I preserve gogo and
protoc versions.

**Comment by douglas-reid** [comment]
Yes. Before I draft a Makefile, I wanted to make sure the general placement was agreeable

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	@echo `protoc --version` > protoc.version
+
+	@mkdir -p protoc-tmp
+
+	# GOOGLEAPIS Generation
+	@mkdir -p protoc-tmp/$(RPC_PATH)
+
+	## Generate google/rpc/status.pb.go
+	@curl -sS $(GOOGLEAPIS_URL)/google/rpc/status.proto -o $(STATUS_PROTO)
+	@gogoreplace 'option go_package = "google.golang.org/genproto/googleapis/rpc/status;status";' '' $(STATUS_PROTO)
```
Comment (kyessenov) [inline Makefile]: Is there really no way to override this without source change?

File: protoc.version [hunk 3]
```diff
@@ -0,0 +1 @@
+libprotoc 3.0.0
```
Comment (kyessenov) [inline protoc.version]: Can you update to 3.5.0?

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+	## Generate google/rpc/code.pb.go
+	@curl -sS $(GOOGLEAPIS_URL)/google/rpc/code.proto -o $(CODE_PROTO)
+	@gogoreplace 'option go_package = "google.golang.org/genproto/googleapis/rpc/code;code";' '' $(CODE_PROTO)
+	@$(PROTOC) $(PROTO_PATH) $(PLUGIN) $(CODE_PROTO)
+
+	## Generate google/rpc/error_details.pb.go
+	@curl -sS $(GOOGLEAPIS_URL)/google/rpc/error_details.proto -o $(ERR_PROTO)
+	@gogoreplace 'option go_package = "google.golang.org/genproto/googleapis/rpc/errdetails;errdetails";' '' $(ERR_PROTO)
+	@$(PROTOC) $(PROTO_PATH) $(PLUGIN) $(ERR_PROTO)
```
Comment (kyessenov) [inline Makefile]: Can you add `google/api/http.proto` and `google/api/annotations.proto`?

Comment (douglas-reid) [reply Makefile]: Mind if we do that in a follow-on PR ?  Adding things is easy, so there is no reason not to do things in small bites.

File: Makefile (hunk 2 above)
Comment (douglas-reid) [reply Makefile]: Not that I have found yet.

File: protoc.version (hunk 3 above)
Comment (douglas-reid) [reply protoc.version]: um, I guess. I wasn't really looking forward to updating my protoc for a PR 😜 .

**Comment by kyessenov** [comment]
I have vague recollection that something changed between 3.0.0 and 3.5.0
that affects gogoproto.

**Comment by douglas-reid** [comment]
OK. I'll update my protoc and increase the min-version check.

**Comment by douglas-reid** [comment]
Updated to 3.5.0. And pinned to exact version of gogo. 

PTAL.

**Pull request merged**